mining.go

```diff
@@ -1499,15 +1514,16 @@ mempoolLoop:
 		// The fraud proof is not checked because it will be filled in
 		// by the miner.
 		_, err = blockchain.CheckTransactionInputs(subsidyCache, tx,
-			nextBlockHeight, blockUtxos, false, server.chainParams)
+			nextBlockHeight, blockUtxos, false,
+			g.blockManager.server.chainParams)
 		if err != nil {
 			minrLog.Tracef("Skipping tx %s due to error in "+
 				"CheckTransactionInputs: %v", tx.Hash(), err)
 			logSkippedDeps(tx, deps)
 			continue
 		}
 		err = blockchain.ValidateTransactionScripts(tx, blockUtxos,
-			scriptFlags, server.sigCache)
+			scriptFlags, g.blockManager.server.sigCache)
```

Comment:
The same applies to the signature cache.  It should be specified as part of `BlkTmplGenerator` and directly accessible in order to break the dependency on server, just like the upstream commit does.

e.g.

```Go
type BlkTmplGenerator struct {
	policy       *mining.Policy
	txSource     mining.TxSource
	sigCache     *txscript.SigCache
	blockManager *blockManager
	timeSource   blockchain.MedianTimeSource
}
```